src/firmware/app/primitives/move_manager.c

```diff
@@ -0,0 +1,224 @@
+#include "firmware/app/primitives/move_manager.h"
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+#include "firmware/app/control/bangbang.h"
+#include "firmware/app/control/control.h"
+#include "firmware/app/control/physbot.h"
+#include "firmware/shared/physics.h"
+#include "firmware/shared/util.h"
+#include "primitive.h"
+#include "shared/constants.h"
+#include "shared/robot_constants.h"
+
+// these are set to decouple the 3 axis from each other
+// the idea is to clamp the maximum velocity and acceleration
+// so that the axes would never have to compete for resources
+#define TIME_HORIZON 0.05f  // s
+
+// The minimum distance away from our destination that we must be if we
+// are going to rotate the bot onto its wheel axis
+// 2 * P_PI * ROBOT_RADIUS = robot circumference, which is approximately
+// how far the bot would have to turn for one full rotation, so we
+// set it a litle larger than that.
+const float APPROACH_LIMIT = 3 * P_PI * ROBOT_RADIUS;
+
+const float PI_2 = P_PI / 2.0f;
+
+typedef struct MoveManagerState
+{
+    float destination[3], end_speed, major_vec[2], minor_vec[2];
+    // We store a wheel index here so we only have to calculate the axis
+    // we want to use when move start is called
+    unsigned optimal_wheel_axes_index;
+    bool slow;
+} MoveManagerState_t;
+DEFINE_PRIMITIVE_STATE_CREATE_AND_DESTROY_FUNCTIONS(MoveManagerState_t)
+
+/**
+ * call from move_start to choose which wheel axis we will be
+ * using for preliminary rotation. The idea is to pick the wheel
+ * axis that will result in the minimum remaining rotation onto
+ * the bot's final destination angle.
+ *
+ * @param dx the global x position of the bot
+ * @param dy the global y position of the bot
+ * @param current_angle the current angle of the bot
+ * @param final_angle the final destination angle
+ * @return the index of the wheel axis to use
+ */
+unsigned choose_wheel_axis(float dx, float dy, float current_angle, float final_angle);
+
+/**
+ * Calculates the rotation time, velocity, and acceleration to be stored
+ * in a PhysBot data container.
+ *
+ * @param pb The data container that has information about major axis time
+ * and will store the rotational information
+ * @param avel The current rotational velocity of the bot
+ */
+void plan_move_rotation(PhysBot* pb, float avel);
+
+/**
+ * builds an array that contains all of the axes perpendicular to
```

Comment:
Nit: Capitalize description to keep it consistent with the others.

Reply:
fixed